observer/obs_conf.go

```diff
@@ -56,12 +57,33 @@ func (c *ObsConf) validateDebugAddr() error {
 	return nil
 }
 
-func (c *ObsConf) makeMonitors() ([]*monitor, []error, error) {
+func (c *ObsConf) makeMonitors(reg *prometheus.Registerer) ([]*monitor, []error, error) {
 	var errs []error
 	var monitors []*monitor
+	collectors := make(map[string]map[string]*prometheus.Collector)
```

Comment:
- The `prometheus.Collector` here doesn't need to be a pointer, we end up having to dereference it when we assert it to a specific metric type during prober construction.
- Nit, the var name could be more descriptive.

```suggestion
	proberSpecificMetrics := make(map[string]map[string]prometheus.Collector)
```